Sources/Helpers/OFFUrlsHelper.swift

```diff
@@ -10,7 +10,7 @@ import UIKit
 
 class OFFUrlsHelper: NSObject {
 
-    static let codeLang = Bundle.main.preferredLocalizations.first ?? "en"
+    static let codeLang = Bundle.main.currentLocalization
```

Comment:
This has been addressed in #747 , which was just approved, and my latest commit has merged that PR in.

Reply:
and we have also some other incorrect behaviours related to the same cause.
Possibly: https://github.com/openfoodfacts/openfoodfacts-ios/issues/684
https://github.com/openfoodfacts/openfoodfacts-ios/issues/662

Reply:
I do not remember it any more, but is the preferredLocalizations not the list that is used by IOS itself (in the preferences). It should be the default behaviour. The language changer is more an advanced feature for demonstration.

Reply:
@aleene 
 Correct, `preferredLocailzations` is the language of the iOS device settings. However, this change has been merged recently in PR #747 , so I think that is the direction we're taking. At least `currentLocailzation` defaults to `preferredLocailzations` if it hasn't been set, and if that fails too, it defaults to "en"
@teolemon and @philippeauriach can correct me if I'm wrong.

Reply:
OK, I am a bit at the loss why it is not working (anymore?). But maybe forgot to test some scenarios.

Not sure where, but I saw a discussion on changing the url. Beware that the prefix might limit a search to products sold in a particular country (have to check whether this is the case).

Reply:
> OK, I am a bit at the loss why it is not working (anymore?). But maybe forgot to test some scenarios.

It was working, in that the in-app webview was opening. The scenario this addresses is only for the hyperlinks to OFF.org, when the user changes the language in the app.

> `preferredLocalizations` ... should be the default behaviour.

It is the default on the app's first launch. But I think I understand your concern now; are you saying the app's language should change when the device's language changes and then again after the user changes it in the app? I believe that was not the case before I edited this code, instead the app's language stays the same even if the user changes the device's language. 

Of course, I could be wrong. I would like to know if the we are using `preferredLocalizations` or `currentLocalization` for the UI strings throughout the app. I'm guessing it's `currentLocalization` because that seems to be changed dynamically when the user changes languages in the app's settings. Please correct me if I'm wrong.

> Not sure where, but I saw a discussion on changing the url. Beware that the prefix might limit a search to products sold in a particular country (have to check whether this is the case).

I don't think this edit affects product search, only the hyperlinks to OFF.org. Are you saying that the OFF website is going to change the url address from "world-(lang).off.org" to just "(lang).off.org" or something like that?